x-pack/plugins/ml/public/data_frame/components/source_index_preview/source_index_preview.tsx

```diff
@@ -0,0 +1,289 @@
+      isExpander: true,
+      render: (item: EsDoc) => (
+        <EuiButtonIcon
+          onClick={() => toggleDetails(item)}
+          aria-label={
+            itemIdToExpandedRowMap[item._id]
+              ? i18n.translate('xpack.ml.dataframe.sourceIndexPreview.rowCollapse', {
+                  defaultMessage: 'Collapse',
+                })
+              : i18n.translate('xpack.ml.dataframe.sourceIndexPreview.rowExpand', {
+                  defaultMessage: 'Expand',
+                })
+          }
+          iconType={itemIdToExpandedRowMap[item._id] ? 'arrowUp' : 'arrowDown'}
+        />
+      ),
+    });
+  }
+
+  if (!loading && tableItems.length === 0) {
+    return (
+      <EuiEmptyPrompt title={<h2>No results</h2>} body={<p>Check the syntax of your query.</p>} />
+    );
+  }
+
+  return (
+    <EuiPanel>
+      <EuiFlexGroup alignItems="center" justifyContent="spaceBetween">
+        <EuiFlexItem grow={false}>
+          <EuiTitle size="xs">
+            <span>
+              {i18n.translate('xpack.ml.dataframe.sourceIndexPreview.sourceIndexPatternTitle', {
+                defaultMessage: 'Source Index {indexPatternTitle}',
+                values: { indexPatternTitle: indexPattern.title },
+              })}
+            </span>
+          </EuiTitle>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiFlexGroup alignItems="center">
+            <EuiFlexItem>
+              {docFieldsCount > MAX_COLUMNS && (
+                <EuiText size="s">
+                  {i18n.translate('xpack.ml.dataframe.sourceIndexPreview.fieldSelection', {
+                    defaultMessage:
+                      'showing {selectedFieldsLength, number} of {docFieldsCount, number} {docFieldsCount, plural, one {field} other {fields}}',
+                    values: { selectedFieldsLength: selectedFields.length, docFieldsCount },
+                  })}
+                </EuiText>
+              )}
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <EuiText size="s">
+                <EuiPopover
+                  id="popover"
+                  button={
+                    <EuiButtonIcon
+                      iconType="gear"
+                      onClick={toggleColumnsPopover}
+                      aria-label={i18n.translate(
+                        'xpack.ml.dataframe.sourceIndexPreview.selectColumnsAriaLabel',
+                        {
+                          defaultMessage: 'Select columns',
+                        }
+                      )}
+                    />
+                  }
+                  isOpen={isColumnsPopoverVisible}
+                  closePopover={closeColumnsPopover}
+                  ownFocus
+                >
+                  <EuiPopoverTitle>
+                    {i18n.translate(
+                      'xpack.ml.dataframe.sourceIndexPreview.selectFieldsPopoverTitle',
+                      {
+                        defaultMessage: 'Select Fields',
```

Comment:
Should use sentence case here I think `Select fields`

Reply:
Done in 6b4b1f4.